### PR TITLE
475 allow workflows from directories

### LINKED ISSUE #475: Allow workflows to be selected from directories as well as tar balls
For ease of use don't require the user to tar up a directory for the cwl. Allow them to use a directory with the cwl files.

### PR DESCRIPTION
**Pull request by aquan9**
This PR allows you to do a bee_client submit and it will automatically do the packaging step if the workflow path is found to be a directory.

### CODE REVIEW

**Comment by aquan9** [comment]
My plan is to include the documentation updates in a separate PR since this one has such an old base.

File: beeflow/client/bee_client.py [hunk 1]
```diff
@@ -167,14 +167,20 @@ def match_short_id(wf_id):
...
            workdir: pathlib.Path = typer.Argument(...,
            help='working directory for workflow containing input + output files',)):
     """Submit a new workflow."""
     if os.path.exists(wf_path):
-        wf_tarball = open(wf_path, 'rb')
+        # Check to see if the wf_path is a tarball or a directory. Run package() if directory
+        if os.path.isdir(wf_path):
+            print("Detected directory instead of packaged workflow. Packaging Directory...")
+            package(wf_path, pathlib.Path("/tmp"))
+            wf_tarball = open(pathlib.Path("/tmp/" + str(wf_path.name) + ".tgz"), 'rb')
```
Comment (pagrubel) [inline beeflow/client/bee_client.py]: Why place the tarball in /tmp. I'd either like to see it placed in the working dir specified by the submit or cwd. Another option would be to put it in bee_workdir/workflows/<wf_id>/.  It ends up there anyway. Right now it is left in /tmp

**Review by pagrubel — CHANGES_REQUESTED** [review]
I'm thinking you should change the location of the tarball so that we don't have to do clean up.

File: beeflow/client/bee_client.py (hunk 1 above)
Comment (pagrubel) [reply beeflow/client/bee_client.py]: @aquan9 After thinking about this using /tmp is fine, if you remove the tar ball once the workflow is submitted

**Comment by jtronge** [comment]
You might be able to use [tempfile.TemporaryDirectory](https://docs.python.org/3/library/tempfile.html#tempfile.TemporaryDirectory).

File: beeflow/client/bee_client.py (hunk 1 above)
Comment (aquan9) [reply beeflow/client/bee_client.py]: I was able to get it to be stored in bee_workdir/archives where I found other tarballs. The wf_id doesn't appear to get generated until after the submission which is too late in the code. I also tried the tempfile idea, but there are too many places where the code might fail where the tmp dir might need to be cleaned up quickly before submission, so the cleanest thing to do seemed to put it in bee_workdir/archives or some other bee_workdir/<directory> if you prefer something else.

**Comment by aquan9** [comment]
I did a "git rebase develop" to pull in the documentation (and everything else that has happened in develop)

File: beeflow/client/bee_client.py (hunk 1 above)
Comment (pagrubel) [reply beeflow/client/bee_client.py]: Thanks looks good. We may want to change the example a bit in the future. I'm going to add CWL to the command section it is such a small change and should have been stated differently to begin.
This all look great! Thanks, @aquan9

**Review by pagrubel — APPROVED** [review]

**Review by pagrubel — CHANGES_REQUESTED** [review]
It seems the integration tests are failing

**Review by pagrubel — APPROVED** [review]
Looks good!

**Pull request merged**